spec/index.bs

```diff
@@ -114,33 +168,73 @@ interface GPUAdapter : GPUObjectBase {
 };
 </script>
 
+{{GPUAdapter}} has the following attributes:
+
+<dl dfn-type=attribute dfn-for=GPUAdapter>
+    : <dfn>\[[adapter]]</dfn>, of type [=adapter=], readonly
+    ::
+        An internal slot holding the [=adapter=] to which this {{GPUAdapter}}
+        refers.
+
+    : <dfn>name</dfn>
+    ::
+        A human-readable name identifying the adapter.
+        The contents are implementation-defined.
+
+    : <dfn>extensions</dfn>
+    ::
+        A {{GPUExtensions}} object which enumerates the extensions supported
+        by the user agent, and whether each extension is supported by the
+        underlying implementation.
+          - If an extension is not supported by the user agent,
+            it will not be present in the object.
+          - If an extension is supported by the user agent, but
+            not by the [=adapter=], it will be `false`.
+          - If an extension is supported by the user agent and
+            by the [=adapter=], it will be `true`.
+</dl>
+
 ### Getting an Adapter ### {#adapter-creation}
 
-<script type=idl>
-[Exposed=Window]
-interface GPU {
-    // May reject with DOMException  // TODO: DOMException("OperationError")?
-    Promise<GPUAdapter> requestAdapter(optional GPURequestAdapterOptions options);
-};
-</script>
+To get a {{GPUAdapter}}, use {{GPU/requestAdapter()}}.
 
 <script type=idl>
 dictionary GPURequestAdapterOptions {
-    GPUPowerPreference? powerPreference;
+    GPUPowerPreference powerPreference;
 };
-</script>
 
-<script type=idl>
 enum GPUPowerPreference {
     "low-power",
     "high-performance"
 };
 </script>
 
-## GPUDevice ## {#device}
+## Devices ## {#devices}
+
+A <dfn dfn>device</dfn> is an arena within which [=internal objects=] are created.
+A [=device=] can be shared across multiple [=realms=].
+It has the following internal attributes:
+
+<dl dfn-type=attribute dfn-for=device>
+    : <dfn>\[[adapter]]</dfn>, of type [=adapter=], readonly
+    ::
+        The [=adapter=] from which this device was created.
+</dl>
+
+When a [=device=] is created, its capabilities are set to those requested by
+the user in {{GPUAdapter/requestDevice()}} (which must be no greater than the
+capabilities specified by the [=adapter=]).
+These capabilities are enforced upon the usage of objects on the [=device=],
+even if the underlying [=adapter=] can support higher capabilities.
+
+### GPUDevice ### {#gpu-device}
+
+A {{GPUDevice}} refers to a [=device=] and
+exposes the capabilities with which the [=device=] was created.
+It is the top-level object through which [=WebGPU interfaces=] are created.
 
 <script type=idl>
-[Exposed=(Window, Worker)]
+[Exposed=(Window, Worker), Serializable]
 interface GPUDevice : EventTarget {
```

Comment:
Should probably be `GPUObjectBase` and have `GPUDeviceDescriptor` have a label. Its `[[device]]` can refer to itself which will make writing validation that all objects in a call belong to the same device easier.